### PR TITLE
Fix segfault that occurs when trying to add precedence to a productio…

### PR DESCRIPTION
**Pull request by mingodad**
…n with a empty RHS, also change the prod_directives repetition in production from '*' to '?'

### CODE REVIEW

**Comment by mingodad** [comment]
Note that we need to regenerate `src/parser.[ch]` too.

File: src/parse.par [hunk 1]
```diff
@@ -692,7 +692,7 @@ ast_node<char*>		:	'=' identifier		[* @@ = pstrdup( strbuf ); *]
 
 production<PROD*>	:	line_number rhs_opt:rhs code_opt_dup:act
 							ast_node
-							prod_directives*
+							prod_directives?
```
Comment (phorward) [inline src/parse.par]: Hello @mingodad,
as already noted [here](/phorward/unicc/tree/master/targets), `prod_directives` should remain to be allowed multiple times. I think the real bugfix of your problem is the one in the next line! Can you please re-check your grammar by using `prod_directives*` still?

File: src/parse.par [hunk 2]
```diff
@@ -712,7 +712,7 @@ production<PROD*>	:	line_number rhs_opt:rhs code_opt_dup:act
 
 rhs_opt<PROD*>		:	rhs
 
-					|	[*	@@ = create_production( parser,
+					|	[*	@@ = current_prod = create_production( parser,
```
Comment (phorward) [inline src/parse.par:715]: This is the real fix for the bug in #22, and I think this is all required. Thanks! :+1:

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @mingodad, thank you for your pull request.
Please re-check your grammar by keeping `prod_directives*` in line 695, the bugfix in line 715 looks good to me!

File: src/parse.par (hunk 1 above)
Comment (mingodad) [reply src/parse.par]: I don't see any mention to `prod_directives` on the link you've pointed out.
As I understand if we have `prod_directives*` then the user can write something as shown bellow, and it doesn't make sense for me:
```
expr:
  expr '*' expr #precedence tok1 #precedence tok2 #precedence tok3 ...;
```
But if we have `prod_directives?` then we'll get a compile time error if the user try to write more than on precedence directive.

Obs: I edited this message to fix a mistake `prod_directives++` => `prod_directives?`

Comment (phorward) [reply src/parse.par]: Hello @mingodad, the syntax should exactly allow for this:
```
expr:
  expr '*' expr #precedence tok1 #precedence tok2 #precedence tok3 ...;
```
When there might be another production directive implemented in future, your use of `prod_directives?` only allows to specify exactly one directive, and specifying e.g.
```
expr:
  expr '*' expr #precedence tok1 #emit mul;
```
won't be allowed.

Therefore please change this line back to `prod_directives*`.

**Pull request merged**